scripts/setup_wasi.sh

```diff
@@ -22,8 +22,17 @@ if ! command -v cargo-wasi &> /dev/null; then
 fi
 
 if [ ! -d $WASI_INSTALL_DIR ]; then
-    WASI_INSTALL_URL="https://github.com/WebAssembly/wasi-sdk/releases/download/wasi-sdk-$WASI_VERSION/wasi-sdk-$WASI_VERSION_FULL-$WASI_OS.tar.gz"
+    WASI_TARBALL="wasi-sdk-$WASI_VERSION_FULL-$WASI_OS.tar.gz"
+    WASI_INSTALL_URL="https://github.com/WebAssembly/wasi-sdk/releases/download/wasi-sdk-$WASI_VERSION/$WASI_TARBALL"
+    OUTFILE="/tmp/$WASI_TARBALL"
+
     echo "Installing WASI SDK $WASI_VERSION_FULL to $WASI_INSTALL_DIR..."
-    curl --retry 2 -L $WASI_INSTALL_URL | tar zxf - -C $WASI_INSTALL_PARENT_DIR
-fi
 
+    if [ ! -f $OUTFILE ]; then
+        echo "Downloading WASI SDK $WASI_VERSION_FULL to $OUTFILE..."
+        curl --retry 2 -L $WASI_INSTALL_URL -o $OUTFILE
+    fi
+    
+    echo "Extracting to $WASI_INSTALL_PARENT_DIR..."
+    tar zxf $OUTFILE -C $WASI_INSTALL_PARENT_DIR
```

Comment:
Might be worth retrying download if extract fails, not critical.

Reply:
Maybe :-)